plugins/source/aws/resources/services/cloudwatch/metrics.go

```diff
@@ -0,0 +1,69 @@
+package cloudwatch
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+	"github.com/cloudquery/cloudquery/plugins/source/aws/client/tableoptions"
+
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
+	"github.com/cloudquery/cloudquery/plugins/source/aws/client"
+	"github.com/cloudquery/plugin-sdk/v3/schema"
+	"github.com/cloudquery/plugin-sdk/v3/transformers"
+)
+
+type metricOutput struct {
+	types.Metric
+	InputJSON tableoptions.CloudwatchListMetricsInput `json:"input_json"`
+
+	getStatsInputs []tableoptions.CloudwatchGetMetricStatisticsInput
+}
+
+func Metrics() *schema.Table {
+	tableName := "aws_cloudwatch_metrics"
+	return &schema.Table{
+		Name:        tableName,
+		Description: `https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_ListMetrics.html`,
+		Resolver:    fetchCloudwatchMetrics,
+		Multiplex:   client.ServiceAccountRegionMultiplexer(tableName, "monitoring"),
+		Transform:   transformers.TransformWithStruct(&metricOutput{}, transformers.WithSkipFields("ResultMetadata"), transformers.WithUnwrapAllEmbeddedStructs()),
+		Columns: []schema.Column{
+			client.DefaultAccountIDColumn(false),
+			client.DefaultRegionColumn(false),
+		},
+		Relations: []*schema.Table{
+			metricStatistics(),
+		},
+	}
+}
+
+func fetchCloudwatchMetrics(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
+	cl := meta.(*client.Client)
+
+	if len(cl.Spec.TableOptions.CloudwatchMetrics) > 0 && !cl.Spec.UsePaidAPIs {
+		cl.Logger().Info().Msg("skipping `aws_cloudwatch_metrics` because `use_paid_apis` is set to false")
```

Comment:
I wonder if this shouldn't both return an error and log an error-level message, because there's a configuration issue if this is the case.

Reply:
Good catch

Reply:
If a user specifies `tables: "*"` there is no way for them to get a "clean" sync. They would have to add `aws_*_custom` to their skip tables

Reply:
@bbernays I think that's fine. If they specify `*` then their intention seems to be to sync this API, so we should tell them when it's not working.

Reply:
(Plus, there is another way to get a clean sync, and that is to enable `use_paid_apis`)

Reply:
I agree, but if the `table_options` are still in experimental stage I am not sure that users should be required to specify anything for them if they are not actively using them...

Reply:
I think getting an error when using `*` is fine. Besides it probably won't be the single error they get ;-)